src/mds/Migrator.cc

```diff
@@ -793,8 +793,8 @@ void Migrator::export_dir(CDir *dir, mds_rank_t dest)
     return;
   }
 
-  if (!dir->inode->is_base() && dir->get_inode()->get_projected_parent_dir()->get_inode()->is_stray() &&
-      dir->get_parent_dir()->get_parent_dir()->ino() != MDS_INO_MDSDIR(dest)) {
+  if (!dir->inode->is_base() && dir->inode->get_projected_parent_dir()->inode->is_stray() &&
+      dir->inode->get_projected_parent_dir()->get_parent_dir()->inode->is_mdsdir()) {
```

Comment:
if stray inode has subtree, we need to export the subtree to stray inode's auth mds. otherwise mds can't purge stray inode . this change breaks this function